src/main/kotlin/de/itemis/mps/gradle/generate/Plugin.kt

```diff
@@ -0,0 +1,92 @@
+package de.itemis.mps.gradle.generate
+
+import org.gradle.api.GradleException
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.tasks.Copy
+import org.gradle.api.tasks.JavaExec
+import java.io.File
+
+
+data class Plugin(
+        var id: String,
+        var path: String
+)
+
+data class Macro(
+        var name: String,
+        var value: String
+)
+
+open class GeneratePluginExtensions {
+    lateinit var mpsConfig: Configuration
+    var mpsLocation: File? = null
+    var plugins: List<de.itemis.mps.gradle.generate.Plugin> = emptyList()
+    var pluginLocation: File? = null
+    var models: List<String> = emptyList()
+    var macros: List<Macro> = emptyList()
+    var projectLocation: File? = File("./mps-prj")
+    var debug = false
+}
+
+open class GenerateMpsProjectPlugin : Plugin<Project> {
+
+    override fun apply(project: Project) {
+        project.run {
+
+            val extension = extensions.create("generate", GeneratePluginExtensions::class.java)
+            val mpsLocation = extension.mpsLocation ?: File(project.buildDir, "mps")
+
+            val genConfig = configurations.maybeCreate("genRuntime")
+            genConfig.isVisible = false
+
+            afterEvaluate {
+                val mpsVersion = extension
+                        .mpsConfig
+                        .resolvedConfiguration
+                        .firstLevelModuleDependencies.find { it.moduleGroup == "com.jetbrains" && it.moduleName == "mps" }
+                        ?.moduleVersion ?: throw GradleException("MPS configuration doesn't contain MPS")
```

Comment:
What if I package and use my RCP instead of MPS here?

Reply:
Se my comment below.